mycli/sqlexecute.py

```diff
@@ -18,17 +20,67 @@
     FIELD_TYPE.NULL: type(None)
 })
 
+
+class ServerSpecies(enum.Enum):
+    MySQL = 'MySQL'
+    MariaDB = 'MariaDB'
+    Percona = 'Percona'
+
+
+class ServerInfo:
+    def __init__(self, species, version_str):
+        self.species = species
+        self.version_str = version_str
+        if species == ServerSpecies.Percona:
+            self.version = int(version_str)
+        else:
+            self.version = self.calc_mysql_version_value(version_str)
+
+    @staticmethod
+    def calc_mysql_version_value(version_str):
+        try:
+            major, minor, patch = version_str.split('.')
+        except ValueError:
+            return 0
+        else:
+            return int(major) * 10_000 + int(minor) * 100 + int(patch)
+
+    @classmethod
+    def from_version_string(cls, version_string):
+        version_re = r'(?P<version>[0-9\.]+)-(?P<comment>[A-Za-z0-9_]+)'
+        match = re.match(version_re, version_string)
+
+        if match is not None:
+            comment = match.group('comment')
+            mysql_version = match.group('version')
+            if 'MariaDB' in comment:
+                species = ServerSpecies.MariaDB
+                parsed_version_str = mysql_version
+            elif comment.isdigit():
+                species = ServerSpecies.Percona
+                parsed_version_str = comment
+            else:
+                species = ServerSpecies.MySQL
+                parsed_version_str = mysql_version
+        else:
+            species = None
```

Comment:
How about adding `ServiceSpecies.Unknown` (or Generic or something like that)? This way, you won't need to check `species` for None.

Reply:
good idea